File: angular15-starter-project-main/src/app/services/decPenalite/dec-penalite.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { environment } from 'src/environnement';

@Injectable({
  providedIn: 'root'
})
export class DecPenaliteService {

  private apiUrl = `${environment.apiUrl}/DecPenalite`;
  private decPenalitesSubject = new BehaviorSubject<any[]>([]);
  decPenalites$ = this.decPenalitesSubject.asObservable();

  constructor(private http: HttpClient) { }

  getDecPenalite(numMarche: number, idTypePen: number, numPieceFourn: number) : Observable<any> {
    const params = new HttpParams()
    .set('numMarche', numMarche)
    .set('idTypePen', idTypePen)
    .set('numPieceFourn', numPieceFourn);
  return this.http.get<any>(this.apiUrl, { params });
  }

  addDecPenalite(data: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}`,data)
  }
}

File: src/main/java/com/afh/gescomp/service/DecPenaliteService.java
package com.afh.gescomp.service;


import com.afh.gescomp.dto.DecPenaliteDTO;
import com.afh.gescomp.model.primary.DecPenalite;

public interface DecPenaliteService {
    DecPenalite saveOrUpdateDecPenalite(DecPenaliteDTO decPenaliteDTOPenalite);
    DecPenalite getDecPenalite(Long numMarche, Long idTypePen, Long numPieceFourn);
}
